File: honoka-sdk-jvm/honoka-utils/src/main/java/de/honoka/sdk/util/ui/UiUtils.java
package de.honoka.sdk.util.ui;

import com.formdev.flatlaf.FlatLightLaf;
import de.honoka.sdk.util.various.CodeUtils;

import javax.swing.*;

public class UiUtils {

    public static void setUiStyleWithCurrentOs() {
        //noinspection CodeBlock2Expr
        CodeUtils.runCatching(() -> {
            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
        });
        FlatLightLaf.setup();
    }
}

File: honoka-sdk-jvm/honoka-kotlin-utils/src/main/java/de/honoka/sdk/util/kotlin/various/UiUtils.kt
package de.honoka.sdk.util.kotlin.various

import de.honoka.sdk.util.ui.ConsoleWindow
import javax.swing.ImageIcon
import javax.swing.JFrame
import javax.swing.UIManager

object UiUtils {

    @Volatile
    private var hasStyle = false

    @Synchronized
    fun setUiStyleWithCurrentOs() {
        if(hasStyle) return
        UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName())
        hasStyle = true
    }
}

fun JFrame.setDefaultIconImage() {
    val url = ConsoleWindow::class.java.getResource("/img/java.png")
    iconImage = ImageIcon(url).image
}
